script/src/verify.rs

```diff
@@ -45,17 +47,17 @@ use core::sync::atomic::{AtomicBool, Ordering};
 mod tests;
 
 pub enum ChunkState {
-    Suspended(Option<ResumableMachine>),
+    Suspended(Vec<ResumableMachine>, Arc<Mutex<MachineContext>>),
     Completed(Cycle),
 }
 
 impl ChunkState {
-    pub fn suspended(machine: ResumableMachine) -> Self {
-        ChunkState::Suspended(Some(machine))
+    pub fn suspended(machines: Vec<ResumableMachine>, context: Arc<Mutex<MachineContext>>) -> Self {
+        ChunkState::Suspended(machines, context)
     }
 
     pub fn suspended_type_id() -> Self {
-        ChunkState::Suspended(None)
+        ChunkState::Suspended(vec![], Default::default())
```

Comment:
In `TransactionScriptsVerifier::verify_group_with_chunk`, if `verifier.verify()` get `Err(ScriptError::ExceededMaximumCycles(_))`, it will return an empty `vms`.

Is is possiable that the empty `vms` received by `TransactionScriptsVerifier::resume_from_snap` and cause unexpected behavior?

Reply:
The answer here has 2 sides:

* The only chance when `ChunkState::Suspended` contains an empty `vms` vector, is cycle limit is reached when we are verifying the embedded type ID script. However at the very top of `verify_group_with_chunk`, we are checking if current script is the embedded type ID script, so when we are entering `resume_from_snap` again, `resume_from_snap` will call `verify_group_with_chunk`, which checks against the embedded type ID script with special handling. So having empty `vms` here won't cause any problem.
* That being said, right now we have `resume_from_state` in addition to `resume_from_snap`, and `resume_from_state` is missing the check for th embedded type ID script, nor the empty `vms`, I've fixed this part. Good catch.

Reply:
Thank you. :heart_decoration: